onedal/primitives/tests/test_kernel_functions.py

```diff
@@ -66,7 +66,8 @@ def test_dense_self_rbf_kernel(queue):
     result = rbf_kernel(X, queue=queue)
     expected = sklearn_rbf_kernel(X)
 
-    assert_allclose(result, expected, rtol=1e-14)
+    tol = 1e-5 if result.dtype == np.float32 else 1e-14
```

Comment:
Yikes, a 10^9 change in performance. Would this also warrant an investigation?

Reply:
I am not sure why 1e-14 was used here specifically - generally we don't go more specific than 1e-7. For rtol, 1e-5 should be acceptable, as this is a typical threshold used in other fp32 testing